queries/migrations.create_prune_galleries_trigger.sql

```diff
@@ -11,17 +11,50 @@ SELECT g.user_id INTO _user_id FROM galleries g
                                         JOIN wallets w ON g.user_id = w.user_id AND w.id = OLD.wallet_id
     LIMIT 1;
 
--- Cleanup pivot table
+-- Mark all gallery nfts as soft deleted if no longer owned by any user
+WITH unassinged_nfts AS (
+    SELECT gallery_id, nft_id FROM galleries
+                                       JOIN nft_gallery ng on galleries.id = ng.gallery_id
+                                       JOIN nfts n on ng.nft_id = n.id AND n.wallet_id IS NULL
+    WHERE user_id = _user_id
+)
+UPDATE nft_gallery
+    SET deleted_at = NOW()
+FROM unassinged_nfts un
+WHERE nft_gallery.gallery_id = un.gallery_id AND nft_gallery.nft_id = un.nft_id;
+
+-- remove gallery nfts relationship if nft belongs to another user or is deleted
 WITH removed_nfts AS (
     SELECT gallery_id, nft_id FROM galleries
                                        JOIN nft_gallery ng on galleries.id = ng.gallery_id
-                                       JOIN nfts n on ng.nft_id = n.id AND (n.wallet_id IS NULL OR n.wallet_id != OLD.wallet_id OR n.deleted_at IS NOT NULL)
+                                       JOIN nfts n on ng.nft_id = n.id AND (n.wallet_id != OLD.wallet_id OR n.deleted_at IS NOT NULL)
     WHERE user_id = _user_id
 )
 DELETE FROM nft_gallery
     USING removed_nfts rm
 WHERE nft_gallery.gallery_id = rm.gallery_id AND nft_gallery.nft_id = rm.nft_id;
 
+
+-- Mark all galleries with only soft deleted nfts as soft deleted
+WITH soft_deleted_nfts AS (
+    SELECT gallery_id
+    FROM nft_gallery
+    WHERE deleted_at IS NOT NULL
+    GROUP BY gallery_id
+    HAVING COUNT(*) = COUNT(deleted_at)
+)
+
+UPDATE galleries
+SET deleted_at = NOW()
+WHERE
+    user_id = _user_id
+AND
+    EXISTS (
+        SELECT 1
+        FROM soft_deleted_nfts
+        WHERE galleries.id = soft_deleted_nfts.gallery_id
+    );
+
```

Comment:
The changes here affect an existing trigger. We need a migration for this or the existing database won't see the changes.

Either we do it manually, or put something like this into a separate migration that runs after the original one:

```sql
BEGIN;

DROP TRIGGER IF EXISTS prune_empty_galleries_on_nft_change on nfts;
CREATE TRIGGER prune_empty_galleries_on_nft_change
    AFTER UPDATE ON nfts
    FOR EACH ROW WHEN ( (OLD.wallet_id IS NOT NULL AND NEW.wallet_id IS NULL) OR (OLD.wallet_id != NEW.wallet_id) OR (OLD.deleted_at IS NULL AND NEW.deleted_at IS NOT NULL))
    EXECUTE PROCEDURE prune_empty_galleries();

COMMIT;
```

Reply:
@oXtxNt9U good catch, [fixed here](https://github.com/ArdentHQ/dashbrd/pull/176/commits/6fc0c9ab5c7c4d260b1a8d465a28dfec2bcf568c)